### PR TITLE
Update Git integration docs

### LINKED ISSUE #360: Update Git provider instructions to cover out of the box and custom configurations in each providers' page

### PR DESCRIPTION
**Pull request by heitortsergent**
Related to #353. Fixes #360.

Update Git integration section and individual docs so instructions for cloud and self-hosted providers live in each provider's docs.

### CODE REVIEW

**Comment by heitortsergent** [comment]
@pamgoodrich I created this PR as a placeholder for the changes I mentioned in #353, so each Git provider doc page would have self-hosted + cloud instructions. If you think this is a good direction I can do some more work on it?

There's more repetition because of these changes, but I think overall it makes for better docs. People can link to the GitLab integration page and it should contain all the information necessary for that integration without having to open multiple links.

**Comment by heitortsergent** [comment]
@chohmann I mentioned this PR on #353, would love to get some feedback from you as well!

One thing I wanna mention is that I added the Firewall/IP Address information to all the Git pages... @pamgoodrich mentioned that you have discussed this once before and decided that it would be better to have it in one place instead of multiple pages. I understand there's maintenance and search implications for having this content duplicated, but I also think it's valuable to have each page be a "full" guide, and have all the information the user might need to set it up without having to go to different links.

What do you think?

File: 2.-workspaces/configure-git/b.bitbucket-cloud.md [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+1. Select **Settings** from the workspace home page.
+2. From the **Integrations** section, select **Add** next to Bitbucket Cloud.
+3. Select **Use Default Configuration,** and then select **Save**.
+
+In the set up dialog you can also customize if you'd like to:
+
+- Enable/disable the integration
+- Allow users to sign in and sign up by using Bitbucket Cloud
+
+## Self-Hosted or Custom Configuration
```
Comment (chohmann) [inline 2.-workspaces/configure-git/b.bitbucket-cloud.md:14]: I actually don't know if Bitbucket Cloud has a self-hosted option 🤔 . Their self hosted option is Bitbucket Server, which is an entirely different product, which is why we have it as an entirely different integration type. You should absolutely double-check me on this, but I believe Bitbucket Cloud is only Cloud and does have on prem.

**Review by chohmann — COMMENTED** [review]
This looks great! Just one comment/question around Bitbucket Cloud and if they even offer self-hosted option. 

As for whether the IP info should be in one location vs. many, I think that's more of a question for you and @pamgoodrich to figure out since it's more of a maintenance concern, and you guys are the ones doing that maintenance.

I do think it's important to have a way to get to that info from all of these pages, but whether that's a link to a shared location, or actually in each page I think it's y'alls decision.

File: 2.-workspaces/configure-git/b.bitbucket-cloud.md (hunk 1 above)
Comment (heitortsergent) [reply 2.-workspaces/configure-git/b.bitbucket-cloud.md:14]: You're right, Bitbucket Server is their self-hosted offering. Thanks for catching this!

File: 2.-workspaces/configure-git/b.bitbucket-cloud.md [hunk 2]
```diff
@@ -1,6 +1,21 @@
 # Bitbucket Cloud
 
-Use these steps to configure your Bitbucket Cloud integration if you have a self-hosted deployment or need custom integration settings.
+Bitbucket Cloud is supported out of the box in Stoplight. To enable your workspace Bitbucket Cloud integration:
+
+1. Select **Settings** from the workspace home page.
+2. From the **Integrations** section, select **Add** next to Bitbucket Cloud.
+3. Select **Use Default Configuration,** and then select **Save**.
+
+In the set up dialog you can also customize if you'd like to:
```
Comment (pamgoodrich) [inline 2.-workspaces/configure-git/b.bitbucket-cloud.md:9]: set up dialog > **Setup** dialog

File: 2.-workspaces/configure-git/c.bitbucket-server.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ Configure your Bitbucket Server integration by following these instructions.
 3. Test your settings.
 4. Select **Save**.
 
+## Firewall and Fixed IP Address Configuration
```
Comment (pamgoodrich) [inline 2.-workspaces/configure-git/c.bitbucket-server.md:14]: I really do struggle with having this in so many places. I'm willing to do it for this area of the doc, but we should minimize this approach in other areas.

File: 2.-workspaces/configure-git/c.bitbucket-server.md [hunk 4]
```diff
@@ -11,6 +11,12 @@ Configure your Bitbucket Server integration by following these instructions.
...
+
+All network traffic from Stoplight originates from a fixed IP address. This is done so that network administrators can allow Stoplight to access your internal Git or identity providers with minimal risk to your network.
+
+If your self-hosted Git provider is behind a firewall, please add our fixed IP address **35.226.194.249** to your allowlist on port 443 over HTTPS.
```
Comment (pamgoodrich) [inline 2.-workspaces/configure-git/c.bitbucket-server.md:18]: Remove "please." https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/please

File: 2.-workspaces/configure-git/e.gitlab.md [hunk 5]
```diff
@@ -1,8 +1,21 @@
 # GitLab
 
+GitLab is supported out of the box in Stoplight. To enable your workspace GitLab integration:
+
+1. Select **Settings** from the workspace home page.
+2. From the **Integrations** section, select **Add** next to GitLab.
+3. Select **Use Default Configuration,** and then select **Save**.
+
+In the set up dialog you can also customize if you'd like to:
```
Comment (pamgoodrich) [inline 2.-workspaces/configure-git/e.gitlab.md:9]: See comment from above.

File: 2.-workspaces/configure-git/e.gitlab.md [hunk 6]
```diff
@@ -1,8 +1,21 @@
...
+- Allow users to sign in and sign up by using GitLab
+
+## Self-Hosted or Custom Configuration
+
 > This feature is available on the Stoplight **Pro** plan and above.
 
-Use these steps to configure your GitLab integration if you have a self-hosted deployment or need custom integration settings.
+By default, the GitLab integration will use Stoplight's OAuth application, but you can also customize your GitLab integration to use your own OAuth app.
 
 ## GitLab Setup
```
Comment (pamgoodrich) [inline 2.-workspaces/configure-git/e.gitlab.md:20]: Same comment as above on heading level.

File: 2.-workspaces/configure-git/b.bitbucket-cloud.md [hunk 7]
```diff
@@ -1,6 +1,21 @@
...
+- Enable/disable the integration
+- Allow users to sign in and sign up by using Bitbucket Cloud
+
+## Custom Configuration
+
+> This feature is available on the Stoplight **Pro** plan and above.
+
+By default, the Bitbucket Cloud integration will use Stoplight's OAuth application, but you can also customize your Bitbucket integration to use your own OAuth app.
 
 ## Bitbucket Cloud Setup
```
[the remark this answers is not in the document]
Comment (heitortsergent) [reply 2.-workspaces/configure-git/b.bitbucket-cloud.md:20]: Ah thanks for catching that. The heading level for that was supposed to be ###.

**Review by pamgoodrich — APPROVED** [review]

**Pull request merged**